source/js/addons.js

```diff
@@ -0,0 +1,183 @@
+
+var addons = [];
+var addonsGameLibrary = [];
+var addonsMetadataProvider = [];
+var addonsGeneric = [];
+var addonsThemeDesktop = [];
+var addonsThemeFullscreen = [];
+
+function sortByName(a, b) {
+    var aName = a.name.toLowerCase();
+    var bName = b.name.toLowerCase();
+    return ((aName < bName) ? -1 : ((aName > bName) ? 1 : 0));
+}
+
+$(document).ready(function () {
+    getAddonList();
+    hideAlerts();
+
+    $("#searchBox").keyup(function (data) {
+        if (data.target.value.length > 0) {
+            var searchResult = search(data.target.value);
+            resortAddons(searchResult.map(function (val) { return val.item }));
+            rerenderAddons();
+        }
+        else {
+            resortAddons(addons);
+            rerenderAddons();
+        }
+    });
+});
+
+function getAddonList() {
+    $.getJSON(apiUrl, function (data) {
+        addons = data.data.sort(sortByName);
+        resortAddons(data.data);
+        rerenderAddons();
+    });
+}
+
+function resortAddons(data) {
+    addonsGameLibrary = [];
+    addonsMetadataProvider = [];
+    addonsGeneric = [];
+    addonsThemeDesktop = [];
+    addonsThemeFullscreen = [];
+    $.each(data, function (_key, val) {
+        switch (val.type) {
+            case 1:
+                addonsMetadataProvider.push(val);
+                break;
+            case 2:
+                addonsGeneric.push(val);
+                break;
+            case 3:
+                addonsThemeDesktop.push(val);
+                break;
+            case 4:
+                addonsThemeFullscreen.push(val);
+                break;
+            case undefined:
+                addonsGameLibrary.push(val);
+                break;
+            default:
+        }
+    });
+}
+
+function rerenderAddons() {
+    renderAddonsSection(addonsMetadataProvider, "addonsMetadataProvider", "Metadata Sources");
+    renderAddonsSection(addonsGeneric, "addonsGeneric", "Generic");
+    renderAddonsSection(addonsThemeDesktop, "addonsThemeDesktop", "Themes Desktop");
+    renderAddonsSection(addonsThemeFullscreen, "addonsThemeFullscreen", "Themes Fullscreen");
+    renderAddonsSection(addonsGameLibrary, "addonsGameLibrary", "Libraries");
+}
+
+function sanitize(input) {
+    var output = input.replace(/<script[^>]*?>.*?<\/script>/gi, '').
+        replace(/<[\/\!]*?[^<>]*?>/gi, '').
+        replace(/<style[^>]*?>.*?<\/style>/gi, '').
+        replace(/<![\s\S]*?--[ \t\n\r]*>/gi, '').
+        replace(/&nbsp;/g, '');
+    return output;
+}
+
+function renderAddonsSection(data, type, name) {
+    var htmlItems = [];
+    if (data.length > 0) {
+        $.each(data, function (_key, val) {
+            htmlItems.push(" \
+            <li class='list-group-item'> \
+                <div class='row'> \
+                    <div class='col-sm-2'> \
+                        " + (val.iconUrl != undefined ? "<img src=" + encodeURI(val.iconUrl) + " width=120 loading='lazy'></img>" : "") + "\
```

Comment:
Avoid unquoted attributes with `encodeURI()`:
```suggestion
                        " + (val.iconUrl != undefined ? '<img src="' + encodeURI(val.iconUrl) + '" width=120 loading="lazy"></img>' : "") + "\
```